### DISCUSSION TITLE
`git@gitlab.com: Permission denied (publickey)`

### DISCUSSION THREAD

**Discussion by Pysis868** [community/community#49959] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

General

### Body

I guess same for me:
https://docs.github.com/en/authentication/troubleshooting-ssh/error-permission-denied-publickey

Just trying `ssh -vvvT git@gitlab.com 2>&1 | less` to make a push work, so no sudo, and using git user.
`debug1: Authenticating to gitlab.com:22 as 'git'`
```
> ls -al ~/.ssh
total #
drwx------. 2 <user> <user> <datetime>  ./
drwx------. 61 <user> <user> <datetime>  ../
-rw-r--r--. 1 <user> <user> <datetime>  authorized_keys
-rw-r--r--. 1 <user> <user> <datetime>  config
-rw-------. 1 <user> <user> <datetime>  id_ecdsa
-rw-r--r--. 1 <user> <user> <datetime>  id_ecdsa.pub
-rw-------. 1 <user> <user> <datetime>  id_rsa
-rw-r--r--. 1 <user> <user> <datetime>  id_rsa.pub
-rw-------. 1 <user> <user> <datetime>  known_hosts
> id
uid=#(<user>) gid=#(<user>) groups=#(<user>),...
```

Reading my and system ssh config, no options statement.
No hosts entry.
`debug1: Connecting to gitlab.com [172.65.251.78] port 22.`

```
> ssh-keygen -l
/home/<me>/.ssh/id_rsa
3072 SHA256:... ... (RSA)
```
Verified matches SSH keys configured in website settings, all with Read/Write (R/W) access.
https://github.com/settings/keys
`ssh-add` added agent log entries, but did not help.

```
debug1: identity file /home/<user>/.ssh/id_rsa type 0
debug1: identity file /home/<user>/.ssh/id_ecdsa type 2
......
debug1: Will attempt key: /home/<user>/.ssh/id_rsa RSA SHA256:...
debug1: Will attempt key: /home/<user>/.ssh/id_ecdsa ECDSA SHA256:...
......
debug1: Offering public key: /home/<user>/.ssh/id_rsa RSA SHA256:...
debug3: send packet: type 50
debug2: we sent a publickey packet, wait for reply
debug3: receive packet: type 51
debug1: Authentications that can continue: publickey
debug1: Offering public key: /home/<user>/.ssh/id_ecdsa ECDSA SHA256:...
debug3: send packet: type 50
debug2: we sent a publickey packet, wait for reply
debug3: receive packet: type 51
debug1: Authentications that can continue: publickey
......
debug1: No more authentication methods to try.
git@gitlab.com: Permission denied (publickey).
```

Re-added keys fresh today, including new ecdsa key.  Also did not help..
Thought I made changes/pushes as recently as 2 weeks ago.  GitHub said for my previous RSA key 4 weeks ago from this computer.
I had moved my drive data recently, which is why I checked permissions too.
No project investigation as I am just testing generic authentication first.
No system proxy.

Is it a problem then I only see `Trying private key` for missing files, or `Offering public key` for the existing ones?

[SSH Debug Level 3 Log Messsages](https://drive.google.com/file/d/15EaTvT6JHR_B5ybbwaxojRsB3a29H6R0/view)

Related:
- https://github.com/orgs/community/discussions?discussions_q=ssh+key+Permission+denied
- https://github.com/orgs/community/discussions/48612
- https://github.com/orgs/community/discussions/47277
Not sure:
- https://github.com/orgs/community/discussions/46057
Not related:
- https://github.com/orgs/community/discussions/40529
- https://github.com/orgs/community/discussions/46416
- https://github.com/orgs/community/discussions/47214
- https://github.com/orgs/community/discussions/42239
- https://github.com/orgs/community/discussions/41222
- https://github.com/orgs/community/discussions/42912
- https://github.com/orgs/community/discussions/36228
More:
- https://askubuntu.com/questions/1088184/what-does-identity-file-type-mean-in-ssh-debug-messages

**Comment by airtower-luna** [comment, marked as answer]
The debug log shows that you're trying to connect to `gitlab.com`. That won't work with keys added to your Git**Hub** account. If you want to make an SSH connection to Git**Lab** you need to add your keys to your account there.

**Comment by Pysis868** [comment]
Can't believe.... thanks.  Probably retained from my terminal history doing something else...

After that, just needed a `git remote set-url --push origin (xout)` since I had the http URL set, and I needed to change to the SSH one for write access, even with SSH keys already working, and `git push -v ...` did not really give me any more information..
